### PR TITLE
Replace interface singleton class with cached functions

### PR DESCRIPTION
**Pull request by joeshannon**
- [x] #570

### CODE REVIEW

File: src/blueapi/service/runner.py [hunk 1]
```diff
@@ -32,51 +35,52 @@ class WorkerDispatcher:
...
     ) -> None:
         self._config = config or ApplicationConfig()
         self._subprocess = None
         self._use_subprocess = use_subprocess
-        self._state = EnvironmentResponse(initialized=False)
+        self._state = EnvironmentResponse(
+            initialized=False,
+        )
+
+    def reload(self):
```
Comment (callumforrester) [inline src/blueapi/service/runner.py:49]: Could: There is a functionality equivalent implementation in `main.py`, should probably just call this.

File: src/blueapi/service/runner.py [hunk 2]
```diff
@@ -32,51 +35,52 @@ class WorkerDispatcher:
...
-                initialized=False, error_message=f"Error configuring blueapi: {e}"
+                initialized=False,
+                error_message=str(e),
             )
-            LOGGER.exception(self._state.error_message)
-            return
-        self._state = EnvironmentResponse(initialized=True)
+            LOGGER.exception(e)
 
     def stop(self):
```
Comment (callumforrester) [inline src/blueapi/service/runner.py:68]: I forget whether we came to any conclusion about whether or not this made `start`/`stop` non-idempotent and whether or not we were okay with that...

**Review by callumforrester — APPROVED** [review]

File: src/blueapi/service/runner.py (hunk 1 above)
Comment (joeshannon) [reply src/blueapi/service/runner.py:49]: Done

File: src/blueapi/service/runner.py (hunk 2 above)
Comment (joeshannon) [reply src/blueapi/service/runner.py:68]: I think for now we are ok with it. There is some risk that if things are called in the wrong order e.g. if `worker()` was called before `setup()` then things might be in an inconsistent state but I think we are ok with this risk for now.

**Pull request merged**